pkg/status/status.go

```diff
@@ -200,3 +227,71 @@ func (s *status) updateStatus() {
 		log.Fatalf("Error %v updating ClusterOperator", err)
 	}
 }
+
+// syncChannelConsumer will listen on the sync channel and update the status'
+// syncs and successfulSyncs fields until the stopCh is closed.
+func syncChannelConsumer(instance *status, stopCh <-chan struct{}) {
+	log.Info("Starting sync consumer")
+	go wait.Until(func() {
+		// Listen on SyncCh and update instance accordingly
+		err := <-syncCh
+		instance.syncs++
+		if err == nil {
+			instance.successfulSyncs++
+		}
+		log.Infof("Successful Syncs / Total Syncs: %d/%d", instance.successfulSyncs, instance.syncs)
+
+	}, 0*time.Second, stopCh)
+}
+
+// monitorClusterStatus updates the clusteroperator's status based on
+// the number of successful syncs / total syncs
+func (s *status) monitorClusterStatus(stopCh <-chan struct{}) {
+	// exit if cluster operator api is not present
+	if s.coAPINotPresent {
+		return
+	}
+
+	go syncChannelConsumer(instance, stopCh)
+	go wait.Until(func() {
+		// create the cluster operator in an initial state if it does not exist
+		if s.clusterOperator == nil {
+			s.setProgressing(fmt.Sprintf("Progressing towards %s", s.version))
+		}
+
+		// wait until the operator has synced enough objects
+		if s.syncs < minSyncsBeforeReporting {
```

Comment:
Why are we waiting to report? The fact that I asked the question implies you need to add a comment here to explain.

Reply:
This `if` statement exists to make sure that we've processed a few syncs before declaring that the operator is available. Also - as the code is currently written t- he status.syncRatio` method could divide by zero.

I will make sure to update the comment.